shared/models/git_repo.rb

```diff
@@ -370,7 +385,8 @@ def clone_synchronously(repo_auth: self.repo_auth)
       logger.debug("[#{self.git_config.id}]: Cloning git repo #{self.git_config.git_url}")
       Git.clone(self.git_config.git_url, self.git_config.id,
                 path: self.git_config.containing_path,
-                recursive: true)
+                recursive: true,
+                depth: 1)
```

Comment:
This makes it so remote branches aren't fetched 🙃

Reply:
Yep, was my bad introducing this in this patch where there's no easy way to get remotes and operate with shallow cloning. My on progress PR #283 will fix this. In the meantime, if this is something is causing you issues, you can revert it. And sorry for the inconvenience